internal/testing/factories/factories.go

```diff
@@ -0,0 +1,50 @@
+package factories
+
+import (
+	"context"
+	"crypto/rand"
+	"crypto/rsa"
+	"errors"
+	"fmt"
+
+	"github.com/RHEnVision/provisioning-backend/internal/models"
+	"golang.org/x/crypto/ssh"
+)
+
+var RSAGenerationError = errors.New("rsa key generation failed")
+
+var nameSequence = 1
+
+// GeneratePubkey generates a pubkey and stores it in the context stub
+func GeneratePubkey(ctx context.Context, pubkey models.Pubkey) (models.Pubkey, error) {
+	var err error
+	if pubkey.AccountID == 0 {
+		pubkey.AccountID = 1
+	}
+	if pubkey.Name == "" {
+		pubkey.Name = nextSequenceName("pubkey")
+	}
+	if pubkey.Body == "" {
+		if pubkey.Body, err = generateRandomRSAPubKey(ctx); err != nil {
+			return pubkey, err
+		}
+	}
+	return pubkey, nil
+}
```

Comment:
The comment for this function does not appear to be correct, I don't see anything being stored "in the context stub".

More importantly, I don't like this function at all - it takes a pubkey as an argument by value, then changes some fields and then returns it by value again. What is the purpose of this function? I can't tell neither from the name nor from the body.

The other function `AddPubkey` has the argument named `template` so it looks like this was the intention, tho I find this weird. This function could easily be designed in a way it takes a reference to pubkey and only setting the Body appropriately.